app/services/dunning_campaigns/update_service.rb

```diff
@@ -103,6 +103,9 @@ def handle_applied_to_organization_update
 
       organization.reset_customers_last_dunning_campaign_attempt
 
+      new_applied_dunning_campaign = dunning_campaign.applied_to_organization ? dunning_campaign : nil
+      organization.default_billing_entity.update(applied_dunning_campaign: new_applied_dunning_campaign)
```

Comment:
```suggestion
      organization.default_billing_entity.update!(applied_dunning_campaign: new_applied_dunning_campaign)
```
lets ensure the billing entity is updated.